relay_test.go

```diff
@@ -160,3 +163,46 @@ func TestErrorFrameEndsRelay(t *testing.T) {
 		assert.Equal(t, ErrCodeBadRequest, se.Code(), "Expected BadRequest error")
 	})
 }
+
+// Trigger a race between receiving a new call and a connection closing
+// by closing the relay while a lot of background calls are being made.
+func TestRaceCloseWithNewCall(t *testing.T) {
```

Comment:
Added more clients making concurrent calls so now it always panics without this change.